go-controller/pkg/util/node_annotations.go

```diff
@@ -464,6 +464,20 @@ func NodeTransitSwitchPortAddrAnnotationChanged(oldNode, newNode *corev1.Node) b
 	return oldNode.Annotations[ovnTransitSwitchPortAddr] != newNode.Annotations[ovnTransitSwitchPortAddr]
 }
 
+func NodeZoneClusterChanged(node1, node2 *corev1.Node) bool {
+	// Check if the annotations have changed.
+	if NodeIDAnnotationChanged(node1, node2) {
+		return true
+	}
+	if NodeGatewayRouterLRPAddrAnnotationChanged(node1, node2) {
+		return true
+	}
+	if NodeTransitSwitchPortAddrAnnotationChanged(node1, node2) {
+		return true
+	}
```

Comment:
`NodeGatewayRouterLRPAddrAnnotationChanged` would not affect layer3 secondary network
`NodeIDAnnotationChanged` and NodeTransitSwitchPortAddrAnnotationChanged` would not only affect remote nodes, but local nodes as well.

Reply:
I think its fine to have these checks.  I don't have a strong opinion.